tftypes/value_json.go

```diff
@@ -16,7 +16,28 @@ import (
 // terraform-plugin-go.  Third parties should not use it, and its behavior is
 // not covered under the API compatibility guarantees. Don't use this.
 func ValueFromJSON(data []byte, typ Type) (Value, error) {
-	return jsonUnmarshal(data, typ, NewAttributePath())
+	return jsonUnmarshal(data, typ, NewAttributePath(), JSONOpts{})
+}
+
+// UnmarshalOpts contains options that can be used to modify the behaviour when
+// unmarshalling. Currently, this only contains a struct for opts for JSON but
+// could have a field for Flatmap in the future.
+type UnmarshalOpts struct {
+	JSONOpts JSONOpts
+}
+
+// JSONOpts contains options that can be used to modify the behaviour when
+// unmarshalling JSON.
+type JSONOpts struct {
```

Comment:
Nit: Can we name this `ValueFromJSONOpts` to match the `ValueFromJSON` function it is associated with? I'm sorry this is pedantic but there's other places that deal with JSON and it could be confusing if we ever needed another JSON options struct like this.

Reply:
Consistency is good :)
Have renamed.